src/pldm_fw.cpp

```diff
@@ -113,77 +242,75 @@ types::PldmPacket
 void PldmFramework::sendPanelFunctionToPhyp(
     const types::FunctionNumber& funcNumber)
 {
-    types::PdrList pdrs =
-        utils::getPDR(phypTerminusID, frontPanelBoardEntityId,
-                      stateIdToEnablePanelFunc, "FindStateEffecterPDR");
+    // Initialising with invalid/unused ID
+    pldm_instance_id_t instance = constants::PLDM_INST_ID_MAX + 1;
 
-    if (pdrs.empty())
+    try
     {
-        std::map<std::string, std::string> additionalData{};
-        additionalData.emplace("DESCRIPTION",
-                               "Empty PDR returned for panel entity id.");
-        utils::createPEL("com.ibm.Panel.Error.HostCommunicationError",
-                         "xyz.openbmc_project.Logging.Entry.Level.Warning",
-                         additionalData);
-        return;
-    }
+        // Get PDR for Panel
+        types::PdrList pdrs =
+            utils::getPDR(phypTerminusID, frontPanelBoardEntityId,
+                          stateIdToEnablePanelFunc, "FindStateEffecterPDR");
 
-    types::Byte instance = getInstanceID();
+        if (pdrs.empty())
+        {
+            throw std::runtime_error(
+                "Empty PDR returned for front panel board entity.");
+        }
 
-    types::PldmPacket packet =
-        prepareSetEffecterReq(pdrs, instance, funcNumber);
+        // Allocate instance ID
+        instance = getInstanceID();
 
-    if (packet.empty())
-    {
-        std::map<std::string, std::string> additionalData{};
-        additionalData.emplace(
-            "DESCRIPTION", "pldm:SetStateEffecterStates request message empty");
-        utils::createPEL("com.ibm.Panel.Error.HostCommunicationError",
-                         "xyz.openbmc_project.Logging.Entry.Level.Warning",
-                         additionalData);
-        return;
-    }
+        // Prepare the message packet
+        types::PldmPacket packet =
+            prepareSetEffecterReq(pdrs, instance, funcNumber);
+
+        if (packet.empty())
+        {
+            throw std::runtime_error(
+                "pldm:SetStateEffecterStates request message empty");
+        }
+
+        // Open PLDM socket based communication
+        int fd = openPldm();
```

Comment:
Not sure if we should continue if the fd is "0" as the "openMctpDemuxTransport" failed? or is it ok to continue?

Reply:
libpldm returns fd as 0 when PLDM_HAS_POLL config is not declared. So keeping panel in sync with libpldm.

https://github.com/openbmc/libpldm/blob/5192e2e26f26ffbbf4256429ea5ae42237e011cc/src/transport/transport.c#L27C19-L27C23